stan/math/rev/functor/map_rect_concurrent.hpp

```diff
@@ -45,10 +45,18 @@ map_rect_concurrent(
   };
 
 #ifdef STAN_THREADS
-  tbb::parallel_for(tbb::blocked_range<std::size_t>(0, num_jobs),
-                    [&](const tbb::blocked_range<size_t>& r) {
-                      execute_chunk(r.begin(), r.end());
-                    });
+  // we must use task isolation as described here:
+  // https://software.intel.com/content/www/us/en/develop/documentation/tbb-documentation/top/intel-threading-building-blocks-developer-guide/task-isolation.html
+  // this is to ensure that the thread local AD tape ressource is
+  // not being modified from a different task which may happen
+  // whenever this function is being used itself in a parallel
+  // context (like running multiple chains for Stan)
+  tbb::this_task_arena::isolate([&] {
```

Comment:
This breaks `rstan` with the current version of TBB in `RcppParallel`:
```bash
StanHeaders/include/stan/math/rev/functor/map_rect_concurrent.hpp:54:25:
error: ‘isolate’ is not a member of ‘tbb::this_task_arena’
   54 |   tbb::this_task_arena::isolate([&] {
      |                         ^~~~~~~
```

Reply:
This is ok with the development version of `RcppParallel`.

Reply:
Will the dev version of RcppParallel hit CRAN any time soon? Can we work around this in an easy way you are aware of?

(If RcppParallel will go on CRAN soonish, then we just wait is what I would say)

Reply:
> Will the dev version of RcppParallel hit CRAN any time soon?

I don't know about the time of the next `RcppParellel` CRAN release.

> Can we work around this in an easy way you are aware of?

Yes, https://github.com/stan-dev/rstan/pull/887/commits/258703275120203bc2d029a16702d421d2735ef1 fixes the R-CMD-check and build of the development version of `rstan`. The next release of `rstan` isn't affected as it depends on Stan `v2.26.1` and Math `v4.0.1`.